src/contracts/core/StrategyManager.sol

```diff
@@ -412,260 +315,82 @@ contract StrategyManager is
         // deposit the assets into the specified strategy and get the equivalent amount of shares in that strategy
         shares = strategy.deposit(token, amount);
 
-        // add the returned shares to the depositor's existing shares for this strategy
-        _addShares(depositor, strategy, shares);
+        // add the returned shares to the staker's existing shares for this strategy
+        _addShares(staker, strategy, shares);
+
+        // Increase shares delegated to operator, if needed
+        delegation.increaseDelegatedShares(staker, strategy, shares);
 
-        emit Deposit(depositor, token, strategy, shares);
+        emit Deposit(staker, token, strategy, shares);
         return shares;
     }
 
     /**
-     * @notice Decreases the shares that `depositor` holds in `strategy` by `shareAmount`.
-     * @param depositor The address to decrement shares from
-     * @param strategyIndex The `strategyIndex` input for the internal `_removeStrategyFromStakerStrategyList`. Used only in the case that
-     * the removal of the depositor's shares results in them having zero remaining shares in the `strategy`
-     * @param strategy The strategy for which the `depositor`'s shares are being decremented
+     * @notice Decreases the shares that `staker` holds in `strategy` by `shareAmount`.
+     * @param staker The address to decrement shares from
+     * @param strategy The strategy for which the `staker`'s shares are being decremented
      * @param shareAmount The amount of shares to decrement
-     * @dev If the amount of shares represents all of the depositor`s shares in said strategy,
-     * then the strategy is removed from stakerStrategyList[depositor] and 'true' is returned. Otherwise 'false' is returned.
+     * @dev If the amount of shares represents all of the staker`s shares in said strategy,
+     * then the strategy is removed from stakerStrategyList[staker] and 'true' is returned. Otherwise 'false' is returned.
      */
     function _removeShares(
-        address depositor,
-        uint256 strategyIndex,
+        address staker,
         IStrategy strategy,
         uint256 shareAmount
     ) internal returns (bool) {
         // sanity checks on inputs
-        require(depositor != address(0), "StrategyManager._removeShares: depositor cannot be zero address");
+        require(staker != address(0), "StrategyManager._removeShares: staker cannot be zero address");
```

Comment:
Can we move this `staker != address(0)` check back up to the function `DelegationManager._removeSharesAndQueueWithdrawal()` or elsewhere in DM? This function `_removeShares()` is only called in a loop so we are doing redundant checks.

Reply:
yes! I think that's a great suggestion, please go ahead and make it if you'd like 👍